components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/JsExecutable.java

```diff
@@ -83,6 +90,18 @@ protected void compile(String scriptSource) {
         } catch (ScriptException e) {
             throw new UUFException("An error occurred while evaluating the JavaScript file '" + absolutePath + "'.", e);
         }
+
+        Set<String> keySet = ((ScriptObjectMirror) engineBindings.get("nashorn.global")).keySet();
+        if (keySet.contains(FUNCTION_ON_GET)) {
+            isOnGetFound = true;
+        }
+        if (keySet.contains(FUNCTION_ON_POST)) {
+            isOnPostFound = true;
+        }
+        if (!isOnGetFound && !isOnPostFound) {
+            throw new UUFException("'onGet' and 'onPost' functions not found. Please implement at least on of them.");
```

Comment:
`Either 'FUNCTION_ON_GET' nor 'FUNCTION_ON_POST' cannot be found in JavaScript file 'absolutePath'. Please implement at least one of them.`